colcon_core/package_augmentation/python.py

```diff
@@ -123,7 +123,9 @@ def create_dependency_descriptor(requirement_string):
     }
 
     requirement = parse_requirement(requirement_string)
-    metadata = {}
+    metadata = {
+        'type': 'python',
```

Comment:
Talking with this offline with @cottsay. I'm a bit reserved with the generic `type` here but if we change these to `package_type` or some less verbose name for the origin of the dependency detection I think it will future proof us against having a `type` alongside other `PROPERTY_type` keys.

Reply:
A reasonable reservation.

Some other options:
- `discoverer`
- `origin`

Reply:
To summarize an offline discussion, `origin` is both more descriptive of what value we're trying to represent here, and has a lower likelihood of colliding with any future metadata.